src/librustc_trans/back/link.rs

```diff
@@ -389,7 +389,9 @@ pub fn link_binary(sess: &Session,
                    outputs: &OutputFilenames,
                    crate_name: &str) -> Vec<PathBuf> {
     let mut out_filenames = Vec::new();
+    let mut dylib = false;
```

Comment:
I'd also be fine with just a `if sess.crate_types.borrow().contains(&config::CrateTypeDylib)` check below (this isn't perf-sensitive or anything).